### PR TITLE
KIALI-2379 Add duration control in traffic pages

### PR DESCRIPTION
**Pull request by israel-hdez**
https://issues.jboss.org/browse/KIALI-2379

This is adding a dropdown to the traffic pages:

image

### CODE REVIEW

File: src/pages/AppDetails/AppDetailsPage.tsx [hunk 1]
```diff
@@ -98,7 +99,7 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
       version: ''
     };
     const restParams = {
-      duration: `${TrafficDetails.defaultDuration}s`,
+      duration: `${MetricsHelper.initDuration({}).duration!}s`,
```
Comment (jotak) [inline src/pages/AppDetails/AppDetailsPage.tsx]: It is a bit weird to use `MetricsHelper.initDuration({}).duration!` here, as this function is really useful to compute some metrics related stuff such as the rate interval that depends on duration, which is of no use here. I think the only piece of code that is interesting to re-use is `MetricsDuration.initialDuration()` that takes duration from URL and/or session storage.

Unless there's something I didn't see, I think it's better to change, to have less coupling with the Metrics-specific stuff.

File: src/pages/AppDetails/AppDetailsPage.tsx [hunk 2]
```diff
@@ -157,11 +158,12 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
               </TabPane>
               <TabPane eventKey="traffic">
                 <TrafficDetails
-                  duration={TrafficDetails.defaultDuration}
+                  duration={MetricsHelper.initDuration({}).duration!}
```
Comment (jotak) [inline src/pages/AppDetails/AppDetailsPage.tsx]: same here

File: src/pages/WorkloadDetails/WorkloadDetailsPage.tsx [hunk 3]
```diff
@@ -115,7 +116,7 @@ class WorkloadDetails extends React.Component<RouteComponentProps<WorkloadId>, W
       version: ''
     };
     const restParams = {
-      duration: `${TrafficDetails.defaultDuration}s`,
+      duration: `${MetricsHelper.initDuration({}).duration!}s`,
```
Comment (jotak) [inline src/pages/WorkloadDetails/WorkloadDetailsPage.tsx]: same

File: src/pages/WorkloadDetails/WorkloadDetailsPage.tsx [hunk 4]
```diff
@@ -216,11 +217,12 @@ class WorkloadDetails extends React.Component<RouteComponentProps<WorkloadId>, W
               </TabPane>
               <TabPane eventKey={'traffic'}>
                 <TrafficDetails
-                  duration={TrafficDetails.defaultDuration}
+                  duration={MetricsHelper.initDuration({}).duration!}
```
Comment (jotak) [inline src/pages/WorkloadDetails/WorkloadDetailsPage.tsx]: same

**Comment by abonas** [comment]
is there a need to specify in the titles the same duration? 
right now there is the dropdown "display last 10 mins" but also "inbound last 10 minutes" - is there a situation where those won't match each other so it has to be specified in the title too?

**Comment by israel-hdez** [comment]
@abonas No real need. I was keeping the original design. But, with the new dropdown, I think it's redundant and the text in the titles can go away

File: src/pages/AppDetails/AppDetailsPage.tsx (hunk 1 above)
Comment (israel-hdez) [reply src/pages/AppDetails/AppDetailsPage.tsx]: You are right. I fixed all references.

**Comment by israel-hdez** [comment]
@abonas The text in titles are removed!

**Comment by abonas** [comment]
LGTM

**Pull request merged**